python/src/server.py

```diff
@@ -0,0 +1,53 @@
+# from typing import Any, List, Literal, Optional, Union
+
+# from pydantic import validator
+
+from konan_sdk.konan_service.services import KonanService
+from konan_sdk.konan_service.models import KonanServiceBaseModel
+from konan_sdk.konan_service.serializers import (
+    KonanServiceBasePredictionRequest, KonanServiceBasePredictionResponse)
+
+
+# TODO: [1] Define your request schema
+class MyPredictionRequest(KonanServiceBasePredictionRequest):
+    # Add your predict endpoint input schema here
+    # Follow the convention of <field_name>: <type_hint>
+    # Check https://pydantic-docs.helpmanual.io/usage/models/ for more info
+
+    # Optionally add validators for your features
+    # Follow the pydantic convention
+    # Check https://pydantic-docs.helpmanual.io/usage/validators/ for more info
+
+    pass
+
+
+# TODO: [2] Define your response schema
+class MyPredictionResponse(KonanServiceBasePredictionResponse):
+    # Add your predict endpoint output schema here
+    # Follow the convention of <field_name>: <type_hint>
+    # Check https://pydantic-docs.helpmanual.io/usage/models/ for more info
+
+    pass
+
+
+class MyModel(KonanServiceBaseModel):
+    def __init__(self):
+        super().__init__()
+        # TODO: [3] Initialize your model
+        # Add logic to initialize your actual model here
+        # Maybe load weights, connect to a database, etc ..
+
+        # For example, the following code will load a model saved as a model.pickle file in the models/ directory
+        # import os
+        # import pickle
+        # self.loaded_model = pickle.load(open(f"{os.environ['MODELS_DIR']}/model.pickle", 'rb'))
+
+    def predict(self, req: MyPredictionRequest) -> MyPredictionResponse:
+        # TODO: [4] Implement your prediction logic
+        # Optionally preprocess the request here
```

Comment:
about preprocessing and post-processing hooks, you can make them as methods and call them in predict at the begging and end respectively, better if you can wrap predict it self in a decorator that does this automatically so that predict has only predict logic.
those hooks base implementation would be to return their input as is, and if User want to do something else custom they can override it

Reply:
This was the original implementation indeed, but @NourhanKhaled suggested we leave that up to the user to implement it inside the `predict()` method. (Less possibility of breaking the pipeline with 1 use-implemented method instead of 3 chained together.)